[docs] Improve the Select docs (#37279)
mui/material-ui: docs/data/base/components/select/select.md

### Multi-select

The Select component lets your users select multiple options from the list.

> In constrast to the single-selection mode, the options popup does not close after an item is selected, allowing users to continue choosing mor e options.

Reviewer: ```suggestion
In contrast to the single-selection mode, the options popup doesn't close after an item is selected, enabling users to continue choosing more options.
```

Set the `multiple` prop to turn on the multi-selection mode.

{{"demo": "UnstyledSelectMultiple.js", "defaultCodeOpen": false}}

Note that in the multiple selection mode, the `value` prop (and `defaultValue`) is an array.

### Controlled select

> Select can be used as an uncontrolled or controlled component:

Reviewer: We we've learned with feedback in the X docs, we should add a callout here to briefly explain this concept. We could copy the text from this PR: https://github.com/mui/mui-x/pull/8926

{{"demo": "UnstyledSelectControlled.js", "defaultCodeOpen": false}}

To set the value of the controlled Select, use the `value` prop.

> In the other case the `defaultValue` can be used to set the initial value of the component.

Author: Yes. I rephrased it a bit.

In any case, if you wish to deselect all values pass `null` to the respective prop.

:::warning

> This is one of the differences with Material UI's Select API.

Reviewer: ```suggestion
This pattern is where Base UI's Select differs from the equivalent [Material UI component](/material-ui/react-select/).
```

> There, an empty string was used to deselect all values.

Reviewer: Rephrasing for present tense and active phrasing.

```suggestion
The Material UI Select takes an empty string to deselect all values.
```

### Multi-select

The Select component lets your users select multiple options from the list.
In contrast to the single-selection mode, the options popup doesn't close after an item is selected, enabling users to continue choosing more options.

Set the `multiple` prop to turn on the multi-selection mode.

{{"demo": "UnstyledSelectMultiple.js", "defaultCodeOpen": false}}

Note that in the multiple selection mode, the `value` prop (and `defaultValue`) is an array.

### Controlled select

Select can be used as an uncontrolled or controlled component.

:::info

- The value is **controlled** when its parent manages it by providing a `value` prop.
- The value is **uncontrolled** when it is managed by the component's own internal state. This state can be initialized using the `defaultValue` prop.

> Learn more about the _Controlled and uncontrolled_ pattern in the [React documentation](https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components).

Reviewer: Very tiny style adjustment for the sake of uniformity with other Core docs with this callout.

```suggestion
Learn more about controlled and uncontrolled components in the [React documentation](https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components).
```

:::

{{"demo": "UnstyledSelectControlled.js", "defaultCodeOpen": false}}

To set the value of the controlled Select, use the `value` prop.
The uncontrolled component accepts the `defaultValue` that can be used to set the initial value.

> In any case, if you wish to deselect all values pass `null` to the respective prop.

Reviewer: ```suggestion
In any case, if you wish to deselect all values, pass `null` to the respective prop.
```

:::warning
This pattern is where Base UI's Select differs from the equivalent [Material UI component](/material-ui/react-select/).
The Material UI Select takes an empty string to deselect all values.

> In Base UI, use `null` to achieve this.

Reviewer: To emphasize that this is something the dev has to do.

```suggestion
In Base UI, you must use `null` to achieve this.
```
